ember.md

```diff
@@ -0,0 +1,461 @@
+    // Code
+  }),
+});
+
+// Bad
+
+import Ember from 'ember';
+import DS from 'ember-data';
+
+const { computed } = Ember;
+
+const {
+  attr,
+  hasMany,
+} = DS;
+
+export default DS.Model.extend({
+  children: hasMany('child'),
+  firstName: attr('string'),
+  lastName: attr('string'),
+
+  fullName: computed('firstName', 'lastName', function _fullName() {
+    // Code
+  }),
+});
+
+```
+
+## Controllers
+
+### Define query params first
+
+For consistency and ease of discover, list your query params first in
+your controller. These should be listed above default values.
+
+### Alias your model
+
+It provides a cleaner code to name your model `user` if it is a user. It
+is more maintainable, and will fall in line with future routable
+components
+
+```javascript
+export default Ember.Controller.extend({
+  user: computed.readOnly('model'),
+});
+```
+
+## Templates
+
+### Avoid using partials
+
+Use components. Partials share scope with the parent view, which makes it hard
+to follow the context. Components will provide a consistent scope. Only use
+partials if you really, really need the shared scope.
+
+### Don't yield `this`
+
+Use the hash helper to yield what you need instead.
+
+```hbs
+{{! Good }}
+{{yield (hash thing=thing action=(action "action"))}}
+
+{{! Bad }}
+{{yield this}}
+```
+
+### Use components in `{{#each}}` blocks
+
+Contents of your each blocks should be a single line, use components
+when more than one line is needed. This will allow you to test the
+contents in isolation via unit tests, as your loop will likely contain
+more complex logic in this case.
+
+```hbs
+{{! Good }}
+{{#each posts as |post|}}
+  {{post-summary post=post}}
+{{/each}}
+
+{{! Bad }}
+{{#each posts as |post|}}
+  <article>
+    <img src={{post.image}} />
+    <h1>{{post.title}}</h2>
+    <p>{{post.summar}}</p>
+  </article>
+{{/each}}
+```
+
+### Always use the `action` keyword to pass actions.
+
+Although it's not strictly needed to use the `action` keyword to pass on
+actions that have already been passed with the `action` keyword once,
+it's recommended to always use the `action` keyword when passing an action
+to another component. This will prevent some potential bugs that can happen
+and also make it more clear that you are passing an action.
+
+```hbs
+{{! Good }}
+{{edit-post post=post deletePost=(action deletePost)}}
+
+{{! Bad }}
+{{edit-post post=post deletePost=deletePost}}
+```
+
+### Ordering static attributes, dynamic attributes, and action helpers for HTML elements
+
+Ultimately, we should make it easier for other developers to read templates.
+Ordering attributes and then action helpers will provide clarity.
+
+```hbs
+{{! Bad }}
+
+<button disabled={{isDisabled}} data-auto-id="click-me" {{action (action click)}} name="wonderful-button" class="wonderful-button">Click me</button>
+```
+
+```hbs
+{{! Good }}
+
+<button class="wonderful-button"
+  data-auto-id="click-me"
```

Comment:
So, if I try to distill the previous recommendations in to rules, here's what I get:

* HTML: Prefer splitting html elements to multiline that have 3 or more attributes, optional within 80 characters.
* Ember Components: Anything more than one attribute should be multiline.

I'm a little torn, but my strongest opinion is that we pick one rule (and it's ok if it is a relaxed rule) for both Ember components and HTML.

@nathancarnes, @maprules1000, @andreafrost since I've had the opportunity to basically choose most of the rules, can I appoint you three the Handlebars multiline decision committee?